### DISCUSSION TITLE
[.NET8 + WebAssembly] Debugging experience still is far from being 'OK'

### DISCUSSION THREAD

**Discussion by DierkDroth** [dotnet/aspnetcore#52289] [General]
Since the first days of .NET + WebAssembly many years back I'm looking forward to finally get a debugging experience which would make it possible to do 'native' development in .NET and WebAssembly.

Unfortunately the debugging experience even got worse with .NET 8 - as just experienced (again):
- starting the debugging sessions takes > 5 times longer than starting the WASM app from VisualStudio without debugging
- as a breakpoint it hit it takes ~30 seconds until VisualStudio finally broke
- removing a breakpoint while debugging just is not effective: although VS no longer would show the breakpoint, it's still triggered
- terminating the debugging sesison could freeze VisualStudio for several minutes until it finally becomes responsive again
- and so on and so on...

Do we 'have to live' with that bad debugging experience going forward - implying in my case that I keep doing the actual development work in WindowsAppSDK - or is there any significant improvement in sight?

My setup: latest VisualStudio, latest Windows 11, latest .NET 8. latest Chrome, state-of-the-art development machine. I'm working on a Platform UNO project.
I activate debugging by enabling this line in my launchSettings.json:
` "inspectUri": "{wsProtocol}://{url.hostname}:{url.port}/_framework/debug/ws-proxy?browser={browserInspectUri}"`

**Comment by RTSTALER** [comment]
I have exactly the same problem.
